src/fairseq2/nn/transformer/encoder.py

```diff
@@ -204,6 +221,19 @@ def forward(
 
         return seqs, padding_mask
 
+    def _drop_iter(self) -> Iterator[Tuple[Module, bool]]:
```

Comment:
I think this logic (and some other pieces in the forward pass) are used in decoder as well. Maybe we could consider adding a base component in the future (that both encoder and decoder would inherit from)